#!/usr/bin/env python
PACKAGE = "offboard"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Pxy_p",    double_t,    0, "Pxy for position PID", 1.1,  0, 5.0)
gen.add("Pz_p",    double_t,    0, "Pz for position PID", 1.1,  0, 5.0)

gen.add("Pxy_v",    double_t,    0, "Pxy for vel PID", 1.5,  0, 5.0)
gen.add("Pz_v",    double_t,    0, "Pz for vel PID", 2.7,  0, 5.0)
gen.add("Ixy_v",    double_t,    0, "Ixy for vel PID", 0.05,  0, 5.0)
gen.add("Iz_v",    double_t,    0, "Iz for vel PID", 0.00,  0, 5.0)
gen.add("Dxy_v",    double_t,    0, "Dxy for vel PID", 0.00,  0, 5.0)
gen.add("Dz_v",    double_t,    0, "Dz for vel PID", 0.00,  0, 5.0)

gen.add("Pxy_a",    double_t,    0, "Pxy for acc PID", 0.001,  0, 5.0)
gen.add("Pz_a",    double_t,    0, "Pz for acc PID", 0.002,  0, 5.0)
gen.add("Ixy_a",    double_t,    0, "Ixy for acc PID", 0.0,  0, 5.0)
gen.add("Iz_a",    double_t,    0, "Iz for acc PID", 0.0,  0, 5.0)
gen.add("Dxy_a",    double_t,    0, "Dxy for acc PID", 0.0,  0, 5.0)
gen.add("Dz_a",    double_t,    0, "Dz for acc PID", 0.0,  0, 5.0)

gen.add("Mz",    double_t,    0, "map acc to thrust", 0.0,  0, 5.0)
gen.add("Mxy",    double_t,    0, "map acc to thrust", 0.0,  0, 10.0)


exit(gen.generate(PACKAGE, "offboard", "PID"))
